rust/lance-encoding/src/encoder.rs

```diff
@@ -116,7 +121,15 @@ pub trait BufferEncoder: std::fmt::Debug + Send + Sync {
     /// This method may receive multiple chunks and should encode them all into
     /// a single EncodedBuffer (though that buffer may have multiple parts).  All
     /// parts will be written to the file as one contiguous block.
-    fn encode(&self, arrays: &[ArrayRef]) -> Result<EncodedBuffer>;
+    fn encode(&self, arrays: &[ArrayRef]) -> Result<(EncodedBuffer, EncodedBufferMeta)>;
+}
+
+pub struct EncodedBufferMeta {
+    pub bits_per_value: u64,
+
+    pub bitpacked_bits_per_value: Option<u64>,
```

Comment:
This feels like an odd property to be at this level.  Should this be `pb::ArrayEncoding` instead?

Reply:
This struct is the return value from `BufferEncoder.encode`, and then we use it to construct the `pb::array_encoding::ArrayEncoding` in the `ValueEncoder`. Is that OK?

https://github.com/lancedb/lance/blob/da28952d47af2ac58c58e6f9aaca3f3c06476771/rust/lance-encoding/src/encodings/physical/value.rs#L260-L283

Reply:
Yes, let's proceed for now.  I this I'll be doing a refactor at some point to introduce the concept of "data layouts" and I can take another look at this then.  Let's leave it as is for now.